plugin/core/registry.py

```diff
@@ -43,11 +44,13 @@ def has_manager(self) -> bool:
 
 class LanguageHandlerDispatcher(object):
 
-    def on_start(self, config_name: str, window: WindowLike) -> bool:
+    def on_start(self, config_name: str, window: WindowLike, folders: List[str], file_path: str) -> List[str]:
         if config_name in client_start_listeners:
-            return client_start_listeners[config_name](window)
+            return folders if client_start_listeners[config_name](window) else []
+        elif config_name in client_can_start_listeners:
+            return client_can_start_listeners[config_name](window, folders, file_path)
```

Comment:
I have flashbacks to #897.

I think it's just not great to have to create a new API that is basically a copy of existing one, just with different arguments. There should be a way to add arguments without breaking the API or some other clean solution for that.

I would also like to have some good way of fixing #897 without introducing a new API that duplicates existing functionality so I'm interested in fixing this properly and in a generic way.

Reply:
Yes, solving the same problem.
I'll comment on #897 to see how far we can get there.